src/Patch/HTMLElementSubclasses.js

```diff
@@ -0,0 +1,69 @@
+import Native from './Native.js';
+import CustomElementInternals from '../CustomElementInternals.js';
+import CEState from '../CustomElementState.js';
+import AlreadyConstructedMarker from '../AlreadyConstructedMarker.js';
+
+/**
+ * @param {!CustomElementInternals} internals
+ */
+export default function(internals) {
+  for (let subclass in Native.HTMLElement_subclasses) {
+    patchElement(`HTML${subclass}Element`, Native.HTMLElement_subclasses[subclass]);
+  }
+
+  function patchElement(elementName, NativeElement) {
+    if (!NativeElement) {
+      return;
+    }
+
+    window[elementName] = (function() {
+      /**
+       * @type {function(new: HTMLElement): !HTMLElement}
+       */
+      function PatchedElement() {
+        // This should really be `new.target` but `new.target` can't be emulated
+        // in ES5. Assuming the user keeps the default value of the constructor's
+        // prototype's `constructor` property, this is equivalent.
+        /** @type {!Function} */
+        const constructor = this.constructor;
+
+        const definition = internals.constructorToDefinition(constructor);
+        if (!definition) {
+          throw new Error('The custom element being constructed was not registered with `customElements`.');
+        }
+
+        const constructionStack = definition.constructionStack;
+
+        if (constructionStack.length === 0) {
+          let element;
+          if (definition.name !== definition.localName) {
+            element = Native.Document_createElement.call(document, definition.localName, {is: definition.name});
```

Comment:
Good point. What I was wanting was the `is` attribute to be set for the customized builtin element. But really I should just do `setAttribute('is', definition.name)`

I have updated this, and also added a test to verify that the `is` attribute gets added when you construct a customized builtin with `new XFoo()`